src/map_io/s2map.cc

```diff
@@ -228,7 +228,8 @@ load_s2mf_section(FileRead& fr, int32_t const width, int32_t const height) {
 		fr.data(dw - width);  // skip the alignment junk
 	}
 	while (y < dh) {
-		fr.data(dw);  // more alignment junk++ y;
+		fr.data(dw);  // more alignment junk
+		++y;
```

Comment:
No idea if this is an actual bug.
I see that code more often in the same file?

These are the original old Setters Maps, do we have some around for testing?

Reply:
Clang-tidy flagged this as "bugprone-infinite-loop" and I think it's right. Please tell me how, with the old code, this loop can ever be exited once we enter it? To me it looks like this `++y` was commented out by accident.

Reply:
I agree with @Noordfrees we should fix this in the way given here.
The only way why this never arose is because nobody never loaded the odd sized maps for which this code is written.
to skip the junks properly the change is necessary I think. This is a very good example for the pros of using a codechecking tool. Otherwise we wouldN#t have found this.

Reply:
I will debug this and get an idea what this is about. But if we do so we must change the other places, too

Reply:
as you already guessed this is for skipping some parts in the S2Maps which we might not need probably.
before this part of the code we look whether the section (according to its header) is bigger then the map_size which could happen with S2 in rare occcasions due to weird map sizes. (but it never happened to me to load such a map as we would have ended in the infinite loop)
the preceding for loop reads the x (width) values of a map line by line regarding tom the height while skipping padded values at the end (padded values exist if the section has more columns then defined in map size due to S2 logic).
After reaching the defined height the while loop skips the rest of the lines. For this it is essential to inkrement the variable y after each chunk is read until it reaches the size of dh.
dw and dh are the sizes of the information in the section while width and height are the sizes of the useful information. 
I hope this helps

Reply:
Checked this, was not reached in my case, lets have it.